framework/src/st_things/things_utils/src/things_time.c

```diff
@@ -0,0 +1,43 @@
+#include <time.h>
+
+#include <things_time.h>
+
+things_result things_time_set(unsigned int second, unsigned int microsecond)
+{
+	if ((int)second < 0 || (int)microsecond < 0) {
+		return THINGS_INVALID_ARGS;
+	}
+
+	struct timespec current_time;
+
+	current_time.tv_sec = second;
+	current_time.tv_nsec = (long int)microsecond * 1000;
+
+	if (clock_settime(CLOCK_REALTIME, &current_time) != 0) {
+		return THINGS_FAIL;
+	}
+
+	return THINGS_SUCCESS;
+}
+
+things_result things_time_info_to_utc_time(things_time_info *time_info, unsigned int *utc_second)
+{
+	if (time_info == NULL || utc_second == NULL) {
+		return THINGS_INVALID_ARGS;
+	}
+
+	struct tm tm;
+
+	tm.tm_sec = time_info->second;
+	tm.tm_min = time_info->minute;
+	tm.tm_hour = time_info->hour;
+	tm.tm_mday = time_info->day;
+	tm.tm_mon = time_info->month;
+	tm.tm_year = time_info->year;
+	tm.tm_wday = time_info->day_of_week;
+	tm.tm_yday = time_info->day_of_year;
+
```

Comment:
1. Here tm_wday and tm_yday are dependent on the flag CONFIG_LIBC_LOCALTIME and CONFIG_TIME_EXTENDED. But you have used unconditionally. This breaks compilation when flags disabled. 
2. Your time_info struct and struct tm are no different. Only one member is missing. If you manage same struct members, you can avoid this function and directly use mktime with typecast in the calling function